src/testExplorer/util.ts

```diff
@@ -1,5 +1,41 @@
-import { Range } from "vscode-languageclient/node";
 import * as vscode from "vscode";
+import { Range } from "vscode-languageclient/node";
+import {
+  MetalsTestItem,
+  MetalsTestItemKind,
+  PackageMetalsTestItem,
+  ProjectMetalsTestItem,
+  SuiteMetalsTestItem,
+  TargetName,
+  TargetUri,
+  TestCaseMetalsTestItem,
+} from "./types";
+
+// prettier-ignore
+export function refineTestItem(test: vscode.TestItem, kind: "project", targetUri: TargetUri, targetName: TargetName): ProjectMetalsTestItem;
+// prettier-ignore
+export function refineTestItem(test: vscode.TestItem, kind: "package", targetUri: TargetUri, targetName: TargetName, parent: vscode.TestItem): PackageMetalsTestItem;
+// prettier-ignore
+export function refineTestItem(test: vscode.TestItem, kind: "suite", targetUri: TargetUri, targetName: TargetName, parent: vscode.TestItem): SuiteMetalsTestItem;
+// prettier-ignore
+export function refineTestItem(test: vscode.TestItem, kind: "testcase", targetUri: TargetUri, targetName: TargetName, parent: vscode.TestItem): TestCaseMetalsTestItem;
+// prettier-ignore
+export function refineTestItem(
+  test: vscode.TestItem,
+  kind: MetalsTestItemKind,
+  targetUri: TargetUri,
+  targetName: TargetName,
+  parent?: vscode.TestItem
+): MetalsTestItem {
+  const casted = test as MetalsTestItem;
```

Comment:
```suggestion
  const cast = test as MetalsTestItem;
```
I think it's `cast -> cast -> cast`